packages/core-data/src/resolvers.js

```diff
@@ -78,3 +82,19 @@ export async function* getThemeSupports() {
 	const index = await apiRequest( { path: '/' } );
 	yield receiveThemeSupportsFromIndex( index );
 }
+
+/**
+ * Requests Upload Permissions from the REST API.
+ */
+export async function* hasUploadPermissions() {
+	const header = await apiRequest( { path: '/wp/v2/media', method: 'OPTIONS' } )
```

Comment:
As best I can tell, the `await` here is unneeded, given how you're using `Promise#then` anyways.

Reply:
Without the await, I believe the value of `header` would be a promise, and then the following attempt to operate on it as a string would fail.